src/lgdo/types/table.py

```diff
@@ -166,9 +172,12 @@ def add_field(
         use_obj_size
             if ``True``, resize the table to match the length of `obj`.
         """
-        if not hasattr(obj, "__len__"):
-            msg = "cannot add field of type"
-            raise TypeError(msg, type(obj).__name__)
+        if not isinstance(obj, LGDO):
+            if isinstance(obj, Mapping):
```

Comment:
the function prototype says the obj can only be an LGDOCollection -- add Mapping[str,LGDO] there?